packages/client/src/connection.ts

```diff
@@ -142,9 +150,16 @@ function addDefaults(options: ConnectionOptions): ConnectionOptionsWithDefaults
  * - Convert {@link ConnectionOptions.tls} to {@link grpc.ChannelCredentials}
  * - Add the grpc.ssl_target_name_override GRPC {@link ConnectionOptions.channelArgs | channel arg}
  * - Add default port to address if port not specified
+ * - Set `Authorization` header based on API key (and not already set explicitly)
  */
 function normalizeGRPCConfig(options?: ConnectionOptions): ConnectionOptions {
   const { tls: tlsFromConfig, credentials, callCredentials, ...rest } = options || {};
+  if (rest.apiKey && !rest.metadata?.['Authorization']) {
```

Comment:
I'd prefer we deal with this incoheencies with a TypeError, just like we do for the tls+credentials case below.

Reply:
We should also forbid the credentials + apiKey use case, as `credentials` will override `metadata['Authorization']`